source/Octopus.Tentacle/Scripts/ScriptExecutorFactory.cs

```diff
@@ -1,19 +1,27 @@
 ﻿using System;
+using Octopus.Tentacle.Kubernetes;
+using Octopus.Tentacle.Scripts.Kubernetes;
 
 namespace Octopus.Tentacle.Scripts
 {
     class ScriptExecutorFactory : IScriptExecutorFactory
     {
         readonly Lazy<LocalShellScriptExecutor> shellScriptExecutor;
+        readonly Lazy<KubernetesJobScriptExecutor> kubernetesJobScriptExecutor;
 
-        public ScriptExecutorFactory(Lazy<LocalShellScriptExecutor> shellScriptExecutor)
+        public ScriptExecutorFactory(Lazy<LocalShellScriptExecutor> shellScriptExecutor, Lazy<KubernetesJobScriptExecutor> kubernetesJobScriptExecutor)
         {
             this.shellScriptExecutor = shellScriptExecutor;
+            this.kubernetesJobScriptExecutor = kubernetesJobScriptExecutor;
         }
 
         public IScriptExecutor GetExecutor()
         {
-            return shellScriptExecutor.Value;
+            return KubernetesConfig.UseJobs switch
```

Comment:
Would it be simpler to push as much of the "if k8s do this" into being container registration concern?

Reply:
yeah, fair call. I'll move this into the module and only register the appropriate executor based on the env var

Reply:
Done in [a2a48b7](https://github.com/OctopusDeploy/OctopusTentacle/pull/690/commits/a2a48b76b42da74a82b8d99115f1ce3d09c49010)

There is no longer a factory and the script executors are now conditionaly registered in the Autofac container